perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -78,9 +79,16 @@ public Resource getResourceByName(PerunSession sess, Vo vo, Facility facility, S
 		return getResourcesManagerImpl().getResourceByName(sess, vo, facility, name);
 	}
 
-	public Resource createResource(PerunSession sess, Resource resource, Vo vo, Facility facility) throws InternalErrorException, FacilityNotExistsException {
-		resource = getResourcesManagerImpl().createResource(sess, vo, resource, facility);
-		getPerunBl().getAuditer().log(sess, "{} created.", resource);
+	public Resource createResource(PerunSession sess, Resource resource, Vo vo, Facility facility) throws InternalErrorException, FacilityNotExistsException, ResourceExistsException {
+		try{
+			getResourcesManagerImpl().getResourceByName(sess, vo, facility, resource.getName());
+			throw new ResourceExistsException("Resource with name \"" + resource.getName() + "\" already exists in " +
+					"facility " + facility + " in VO " + vo + ".");
+		} catch (ResourceNotExistsException e) {
+			resource = getResourcesManagerImpl().createResource(sess, vo, resource, facility);
```

Comment:
@licehammer It is ok to do it this way? (I mean doing the logic in catch block after the exception is thrown) I don't like it but it seemed to me as the easiest way, because the method getResourceByName already existed, but it throws exception when resource with that name not exists.